lib/pbench/test/unit/common/test_config.py

```diff
@@ -1,68 +1,252 @@
+        with pytest.raises(BadConfig) as exc:
+            config = PbenchConfig("pbench.cfg")
+            assert config.logger_type == "hostport"
+            assert exc == "No option 'logger_host' in section: 'logging'"
+
+    def test_logger_type_host_value(self, monkeypatch):
+        """verify `host` value of logger_type `hostport`"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {
+                    "logging": {"logger_type": "hostport", "logger_host": ""}
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        with pytest.raises(ValueError) as exc:
+            config = PbenchConfig("pbench.cfg")
+            assert config.logger_type == "hostport"
+            assert exc == "No option 'logger_host' in section: 'logging'"
+
+    def test_logger_type_port(self, monkeypatch):
+        """verify `logger_port` for logger_type `hostport`"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {
+                    "logging": {"logger_type": "hostport", "logger_host": "localhost"}
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        with pytest.raises(BadConfig) as exc:
+            config = PbenchConfig("pbench.cfg")
+            assert config.logger_type == "hostport"
+            assert config.logger_host == "logger.example.com"
+            assert exc == "No option 'logger_port' in section: 'logging'"
+
+    def test_logger_type_port_value(self, monkeypatch):
+        """verify `port` value of logger_type `hostport`"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {
+                    "logging": {
+                        "logger_type": "hostport",
+                        "logger_host": "logger.example.com",
+                        "logger_port": "",
+                    }
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        with pytest.raises(ValueError) as exc:
+            config = PbenchConfig("pbench.cfg")
+            assert config.logger_type == "hostport"
+            assert config.logger_host == "logger.example.com"
+            assert exc == "No option 'logger_port' in section: 'logging'"
```

Comment:
It strikes me that all of the "hostport" failure test scenarios could be combined into a single, parametrize'd test, with input parameters for the sections, the expected exception, and the expected exception message.

Reply:
it seems a little cluttery but sure